src/platform/packages/shared/kbn-data-grid-in-table-search/src/in_table_search_control.tsx

```diff
@@ -0,0 +1,217 @@
+  onChangeToExpectedPage: (pageIndex: number) => void;
+}
+
+export const InTableSearchControl: React.FC<InTableSearchControlProps> = ({
+  pageSize,
+  getColumnIndexFromId,
+  scrollToCell,
+  shouldOverrideCmdF,
+  onChange,
+  onChangeCss,
+  onChangeToExpectedPage,
+  ...props
+}) => {
+  const { euiTheme } = useEuiTheme();
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const shouldReturnFocusToButtonRef = useRef<boolean>(false);
+  const [isInputVisible, setIsInputVisible] = useState<boolean>(false);
+
+  const onScrollToActiveMatch: UseFindMatchesProps['onScrollToActiveMatch'] = useCallback(
+    ({ rowIndex, columnId, matchIndexWithinCell }) => {
+      if (typeof pageSize === 'number') {
+        const expectedPageIndex = Math.floor(rowIndex / pageSize);
+        onChangeToExpectedPage(expectedPageIndex);
+      }
+
+      // The cell border is useful when the active match is not visible due to the limited cell height.
+      onChangeCss(css`
+        .euiDataGridRowCell[data-gridcell-row-index='${rowIndex}'][data-gridcell-column-id='${columnId}'] {
+          &:after {
+            content: '';
+            z-index: 2;
+            pointer-events: none;
+            position: absolute;
+            inset: 0;
+            border: 2px solid ${ACTIVE_HIGHLIGHT_COLOR} !important;
+            border-radius: 3px;
+          }
+          .${HIGHLIGHT_CLASS_NAME}[${CELL_MATCH_INDEX_ATTRIBUTE}='${matchIndexWithinCell}'] {
+            background-color: ${ACTIVE_HIGHLIGHT_COLOR} !important;
+          }
+        }
+      `);
+
+      // getting rowIndex for the visible page
+      const visibleRowIndex = typeof pageSize === 'number' ? rowIndex % pageSize : rowIndex;
+
+      scrollToCell({
+        rowIndex: visibleRowIndex,
+        columnIndex: getColumnIndexFromId(columnId),
+        align: 'center',
+      });
+    },
+    [getColumnIndexFromId, scrollToCell, onChangeCss, onChangeToExpectedPage, pageSize]
+  );
+
+  const {
+    matchesCount,
+    activeMatchPosition,
+    isProcessing,
+    goToPrevMatch,
+    goToNextMatch,
+    renderCellsShadowPortal,
+    resetState,
+  } = useFindMatches({ ...props, onScrollToActiveMatch });
+
+  const showInput = useCallback(() => {
+    setIsInputVisible(true);
+  }, [setIsInputVisible]);
+
+  const hideInput = useCallback(
+    (shouldReturnFocusToButton: boolean = false) => {
+      setIsInputVisible(false);
+      resetState();
+      shouldReturnFocusToButtonRef.current = shouldReturnFocusToButton;
+    },
+    [setIsInputVisible, resetState]
+  );
+
+  // listens for the cmd+f or ctrl+f keydown event to open the input
+  useEffect(() => {
+    const handleGlobalKeyDown = (event: KeyboardEvent) => {
+      if (
+        (event.metaKey || event.ctrlKey) &&
+        event.key === 'f' &&
+        shouldOverrideCmdF(event.target as HTMLElement)
+      ) {
+        event.preventDefault(); // prevent default browser find-in-page behavior
```

Comment:
Jul I cant make it work 🤔 Am I doing something wrong?

<img width="2520" alt="image" src="https://github.com/user-attachments/assets/2d88b24e-d6e0-4f5b-95c5-68826b8ed47f" />

I see the browser find functionality instead

Reply:
@stratoula We keep the default behaviour too. Only if one of grid cells was already in focus and user presses the shortcut, then it will open the in-table search input.

Reply:
Ah I see, got it thanx 👍